.github/workflows/ruff_linter.yml

```diff
@@ -0,0 +1,26 @@
+name: Ruff 
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.12
+
+    - name: Install Ruff
+      run: |
+        python -m pip install --upgrade pip
+        pip install ruff
```

Comment:
Pinning a version of Python libs (i.e., `ruff==x.x.x`) would be a good idea. As well as regularly updating them, this way we can isolate and understand the difference between lib versions.